frontend/src/app/main/ui/ds/utilities/swatch.scss

```diff
@@ -0,0 +1,91 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) KALEIDOS INC
+
+@use "../_borders.scss" as *;
+@use "../_sizes.scss" as *;
+@use "../colors.scss" as *;
+
+.swatch {
+  --border-color: var(--color-accent-primary-muted);
+  --border-radius: #{$br-4};
+  --border-color-active-inset: var(--color-foreground-primary);
+  --border-color-active: var(--color-background-primary);
+
+  --checkerboard-background: repeating-conic-gradient(lightgray 0% 25%, white 0% 50%);
+  --checkerboard-size: 0.5rem 0.5rem;
+
+  border: $b-1 solid var(--border-color);
+  border-radius: var(--border-radius);
+  overflow: hidden;
+
+  & > div {
+    block-size: 100%;
+  }
+
+  &:focus {
+    --border-color: var(--color-accent-primary);
+  }
+}
+
+.small {
+  inline-size: $sz-16;
+  block-size: $sz-16;
+}
+
+.medium {
+  inline-size: $sz-24;
+  block-size: $sz-24;
+
+  .swatch-opacity {
+    --checkerboard-size: 1rem 1rem;
+  }
```

Comment:
Please do not nest CSS selectors, it creates unnecessary specificity. See [our UI Guidelines](https://help.penpot.app/technical-guide/developer/ui/#favor-lower-specificity).